### PR TITLE
[udp,tcp]: introduce preserve original event option

### PR DESCRIPTION
**Pull request by pkoutsovasilis**
## Proposed commit message

This PR introduces the `preserve_event_original` option for `udp` and `tcp` integrations. This is especially useful when a user has enabled the "Syslog Parser" toggle as there's no way to retain the original message because the [syslog](https://www.elastic.co/guide/en/beats/filebeat/current/syslog.html) processor will overwrite message

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

N/A

## How to test this PR locally

```
cd packages/tcp
elastic-package test system -v
```

```
cd packages/udp
elastic-package test system -v
```

## Related issues

- Closes https://github.com/elastic/integrations/issues/10455

## Screenshots

tcp

udp

### CODE REVIEW

File: packages/udp/data_stream/generic/agent/stream/udp.yml.hbs [hunk 1]
```diff
@@ -18,24 +18,48 @@ keep_null: {{keep_null}}
...
+tags:
+  - preserve_original_event
 {{/if}}
 {{#contains "forwarded" tags}}
 publisher_pipeline.disable_host: true
 {{/contains}}
 {{#if processors}}
 processors:
+{{#if preserve_original_event}}
+- copy_fields:
```
Comment (andrewkroh) [inline packages/udp/data_stream/generic/agent/stream/udp.yml.hbs:37]: I was originally thinking that this part would be always-on for the sake of simplicity, but by making it conditional we transfer less data over the wire when the feature is off. 👍

**Review by andrewkroh — APPROVED** [review]

**Comment by pkoutsovasilis** [comment]
@andrewkroh any idea how to deal with coverage issue? I have added test cases for syslog+preserve_original_event and preserve_original_event alone so I am not sure what else I can do to increase it 😄

File: packages/udp/data_stream/generic/agent/stream/udp.yml.hbs (hunk 1 above)
Comment (pkoutsovasilis) [reply packages/udp/data_stream/generic/agent/stream/udp.yml.hbs:37]: cheers 🎉

**Comment by andrewkroh** [comment]
This looks like a problem with that's occurring on several PRs. It seems like the changelog.yml and manifest.yml are being scored with 0% coverage bringing down overall value. 

There was an email about updating PRs with main to address Sonarqube failures. Give that a try that.

**Comment by andrewkroh** [comment]
You can merge this according to https://github.com/elastic/integrations/pull/10225#issuecomment-2210528698

**Pull request merged**